app/code/Magento/NewsletterApi/Api/NewsletterSubscriptionRepositoryInterface.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\NewsletterApi\Api;
+
+/**
+ * Newsletter Subscription CRUD interface.
+ *
+ * @api
+ */
+interface NewsletterSubscriptionRepositoryInterface
+{
+    /**
+     * Save newsletter subscription.
+     *
+     * @param \Magento\NewsletterApi\Api\Data\NewsletterSubscriptionInterface $newsletterSubscription
+     * @return \Magento\NewsletterApi\Api\Data\NewsletterSubscriptionInterface
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function save($newsletterSubscription);
+
+    /**
+     * Retrieve newsletter subscription.
+     *
+     * @param int $newsletterSubscriptionId
+     * @return \Magento\NewsletterApi\Api\Data\NewsletterSubscriptionInterface
+     * @throws \Magento\Framework\Exception\LocalizedException
```

Comment:
* @throws \Magento\Framework\Exception\NoSuchEntityException
should be used instead